solr/core/src/java/org/apache/solr/handler/component/WrappedHttpShardHandlerFactory.java

```diff
@@ -0,0 +1,21 @@
+package org.apache.solr.handler.component;
+
+/**
+ * Wrapping a HttpShardHandlerFactory to avoid many instances of HttpShardHandlerFactory created by
+ * each SolrCore
+ */
+public class WrappedHttpShardHandlerFactory extends ShardHandlerFactory {
```

Comment:
We need configure WrappedHttpShardHandlerFactor now?

Reply:
Yes. it will be in the solrconfig.xml of _FS7 (as the default). We will need to change the mn PR after this is merged to 9_7 and fs-solr :)